extensions/sql-database-projects/src/tools/netcoreTool.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as os from 'os';
+import { isNullOrUndefined } from 'util';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+export const DBProjectConfigurationKey: string = 'sqlDatabaseProjects';
+export const NetCoreInstallLocationKey: string = 'netCoreSDKLocation';
+export const NetCoreInstallationConfirmation: string = localize('sqlDatabaseProjects.NetCoreInstallationConfirmation', "The .NET Core SDK cannnot be located. Project build will not work. Please install the same or update the .Net Core SDK location in settings if already installed.");
+export const UpdateNetCoreLocation: string = localize('sqlDatabaseProjects.UpdateNetCoreLocation', "Update .Net Core location");
+export const InstallNetCore: string = localize('sqlDatabaseProjects.InstallNetCore', "Install .Net Core SDK");
+
+export class NetCoreTool {
+
+	public findOrInstallNetCore(): void {
+		if (!this.isNetCoreInstallationPresent) {
+			this.showInstallDialog();
+		}
+	}
+
+	private showInstallDialog(): void {
+		vscode.window.showInformationMessage(NetCoreInstallationConfirmation, UpdateNetCoreLocation, InstallNetCore).then(async (result) => {
```

Comment:
can you make this be async/await instead of then-ed? :) preferred in TS

Reply:
not sure on this one actually - we follow then pretty extensively. so will keep this